integration/client/imagerules/imageroleauthorizations_test.go

```diff
@@ -363,4 +364,227 @@ func TestImageRoleAuthorizations(t *testing.T) {
+			},
+		}},
+	})
+
+	app = createWaitLoop(ctx, nappinstance)
+
+	require.Equal(t, 1, len(app.Status.Staged.ImagePermissionsDenied), "should have 1 denied permission (rootapp)")
+
+	rmapp(ctx, app)
+}
+
+func TestImageRoleAuthorizationConsumerPerms(t *testing.T) {
+	helper.StartController(t)
+
+	ctx := helper.GetCTX(t)
+	c, _ := helper.ClientAndProject(t)
+	kclient := helper.MustReturn(kclient.Default)
+
+	// enable image role authorizations in acorn config
+	helper.EnableFeatureWithRestore(t, ctx, kclient, profiles.FeatureImageRoleAuthorizations)
+
+	// Delete any existing rules from this project namespace
+	err := kclient.DeleteAllOf(ctx, &internaladminv1.ImageRoleAuthorizationInstance{}, cclient.InNamespace(c.GetNamespace()))
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = kclient.DeleteAllOf(ctx, &internaladminv1.ClusterImageRoleAuthorizationInstance{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Build Image
+	image, err := c.AcornImageBuild(ctx, "./testdata/serviceconsumer/Acornfile", &client.AcornImageBuildOptions{
+		Cwd: "./testdata/serviceconsumer/",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Prep: Create a role that we can use later
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test:get:secrets",
+			Namespace: c.GetNamespace(),
+		},
+		Rules: []rbacv1.PolicyRule{{
+			APIGroups: []string{""},
+			Verbs:     []string{"get"},
+			Resources: []string{"secrets"},
+		}},
+	}
+
+	err = apply.Ensure(ctx, kclient, role)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Integration tests don't have proper privileges so we will by pass the permission validation
+	appInstance := &internalv1.AppInstance{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-consumer",
+			Namespace: c.GetProject(),
+		},
+		Spec: internalv1.AppInstanceSpec{
+			Image: image.ID,
+			GrantedPermissions: []internalv1.Permissions{{
+				ServiceName: "producer.default",
+				Rules: []internalv1.PolicyRule{{
+					PolicyRule: rbacv1.PolicyRule{
+						APIGroups: []string{""},
+						Verbs:     []string{"get"},
+						Resources: []string{"secrets"},
+					},
+				}},
+			}},
+		},
+		Status: internalv1.AppInstanceStatus{},
+	}
+	if err := apply.Ensure(ctx, kclient, appInstance); err != nil {
+		t.Fatal(err)
+	}
+
+	app, err := c.AppGet(ctx, appInstance.Name)
+	if err != nil {
+		t.Fatal(err)
+	}
+	app = helper.WaitForObject(t, helper.Watcher(t, c), &apiv1.AppList{}, app, func(obj *apiv1.App) bool {
+		return obj.Status.AppStatus.Services["producer"].ServiceAcornName != ""
+	})
+
+	serviceAcorn, err := c.AppGet(ctx, app.Status.AppStatus.Services["producer"].ServiceAcornName)
+	require.NoError(t, err, "should not error while getting service acorn")
+
+	serviceAcorn = helper.WaitForObject(t, helper.Watcher(t, c), &apiv1.AppList{}, serviceAcorn, func(obj *apiv1.App) bool {
+		return obj.Status.Staged.PermissionsChecked
+	})
+
+	require.Equal(t, 1, len(serviceAcorn.Status.Staged.ImagePermissionsDenied), "should have 0 denied permissions")
```

Comment:
Bit confused by this. Shouldn't it be requiring that the len of ImagePermissionsDenied is 0, not 1? Or is the error message incorrect?

Reply:
Yeah, the error message was wrong :D